canonical/ubuntu-advantage-client
# [wip] Postinst dont setup beta

Needs integration test fixes and might need python requirements.txt work.

## Proposed Commit Message

    d/postinst: configure esm on any LTS release avoid beta services
    
    When installing ubuntu-advantage-tools on an unattached system,
    enable unauthorized visibility to esm-infra APT sources to allow
    APT to see package availability for security fixes.
    
    One a machine is attached to ubuntu-advantage with a contract token,
    authorized credentials will allow APT to automatically upgrade ESM
    packages.
    
    Avoid laying down esm-apps APT source files if ESM Apps is still
    considered a beta service.


## Test Steps
```bash
lxc launch ubuntu-daily:xenial dev-x
dpkg-buildpackage -us -uc
dpkg -i ../ubuntu-advantage-tools*deb
find /etc/apt | grep esm
apt update
apt policy | grep esm   # expect to see just esm-infra -32768
apt remove --purge ubuntu-advantage-tools

# setup allow_beta : true
mkdir /etc/ubuntu-advantage
cat > /etc/ubuntu-advantage/uaclient.conf <<EOF
# Ubuntu-Advantage client config file.
contract_url: 'https://contracts.canonical.com'
security_url: 'https://ubuntu.com/security'
data_dir: /var/lib/ubuntu-advantage
log_level: debug
log_file: /var/log/ubuntu-advantage.log
features:
  allow_beta: true
EOF

# reinstall with allow_beta setting to trigger laying down esm-apps 
dpkg -i ../ubuntu-advantage-tools*deb
find /etc/apt | grep esm 
apt update 
apt policy | grep esm   # expect to see esm-apps and esm-infra -32768

```


## Desired commit type::


## Checklist:
